Engines/Wine/Settings/always offscreen/script.js

```diff
@@ -4,45 +4,50 @@ include("engines.wine.plugins.regedit");
 /**
  * setting to set always offscreen
 */
-var settingImplementation = {
-    _options: [tr("Default"), tr("Disabled"), tr("Enabled")],
-    // values which are written into the registry, do not translate!
-    _registryValues: ["", "disabled", "enabled"],
-    getText: function () {
+class AlwaysOffscreenSetting {
+    constructor() {
+        this.options = [tr("Default"), tr("Disabled"), tr("Enabled")];
+        // values which are written into the registry, do not translate!
+        this.registryValues = ["", "disabled", "enabled"];
+    }
+
+    getText() {
         return tr("Always offscreen");
-    },
-    getOptions: function () {
-        return this._options;
-    },
-    getCurrentOption: function (container) {
-        var currentValue = new Wine()
+    }
+
+    getOptions() {
+        return this.options;
+    }
+
+    getCurrentOption(container) {
+        const currentValue = new Wine()
             .prefix(container)
             .regedit()
             .fetchValue(["HKEY_CURRENT_USER", "Software", "Wine", "Direct3D", "AlwaysOffscreen"]);
+
         // find matching option (use default if not found)
-        var index = Math.max(this._registryValues.indexOf(currentValue), 0);
-        return this._options[index];
-    },
-    setOption: function (container, optionIndex) {
+        const index = Math.max(this.registryValues.indexOf(currentValue), 0);
+
+        return this.options[index];
+    }
+
+    setOption(container, optionIndex) {
```

Comment:
Similar to `getCurrentOption` this method could be changed to a normal JavaScript setter method if we move the container argument to the constructor:

```javascript
set currentOption(optionIndex) {
   ...
}
```

Reply:
Same like above: what would be the benefits?